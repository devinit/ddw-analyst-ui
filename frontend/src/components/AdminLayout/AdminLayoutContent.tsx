import * as React from 'react';

export const AdminLayoutContent: React.SFC = props => {
  return (
    <React.Fragment>
      { props.children }
    </React.Fragment>
  );
};
