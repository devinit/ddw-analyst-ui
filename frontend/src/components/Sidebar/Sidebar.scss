@media (max-width: 991px) {
  .off-canvas-sidebar nav .navbar-collapse:before, .sidebar:before {
    background-color: inherit;
  }
}
