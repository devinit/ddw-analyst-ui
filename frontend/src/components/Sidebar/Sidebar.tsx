import * as React from 'react';
import { SidebarContent } from '../SidebarContent';
import { SidebarItem } from '../SidebarItem';
import { SidebarLink } from '../SidebarLink';
import { SidebarLogo } from '../SidebarLogo';

interface SidebarProps {
  dataColour?: 'rose' | 'purple' | 'azure' | 'danger' | 'green' | 'orange';
  backgroundColour?: 'black' | 'white' | 'red';
}

export class Sidebar extends React.Component<SidebarProps> {
  static defaultProps: Partial<SidebarProps> = {
    dataColour: 'rose',
    backgroundColour: 'black'
  };
  static Logo = SidebarLogo;
  static Content = SidebarContent;
  static Item = SidebarItem;
  static Link = SidebarLink;

  render() {
    return (
      <div
        className="sidebar"
        data-color={ this.props.dataColour }
        data-background-color={ this.props.backgroundColour }
      >
        { this.renderContent(SidebarLogo) }
        { this.renderContent(SidebarContent) }
      </div>
    );
  }

  private renderContent(type: typeof SidebarLogo | typeof SidebarContent) {
    return React.Children.map(this.props.children, child => {
      if (React.isValidElement(child) && child.type === type) {
        return child;
      }
    });
  }
}
