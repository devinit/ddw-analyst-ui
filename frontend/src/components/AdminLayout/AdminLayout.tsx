import * as React from 'react';
import { Sidebar } from './Sidebar';

interface AdminLayoutProps {
  loading: boolean;
}

export class AdminLayout extends React.Component<AdminLayoutProps> {
  static defaultProps: Partial<AdminLayoutProps> = {
    loading: true
  };

  render() {
    if (this.props.loading) {
      return <div>Loading ...</div>;
    }

    return (
      <div className="wrapper">
        { this.renderContent(Sidebar) }
        <div className="main-panel">
          <div className="content">
            <div className="container-fluid">
              Main Layout
            </div>
          </div>
        </div>
      </div>
    );
  }

  private renderContent(type: typeof Sidebar) {
    return React.Children.map(this.props.children, child => {
      if (React.isValidElement(child) && child.type === type) {
        return child;
      }
    });
  }
}
