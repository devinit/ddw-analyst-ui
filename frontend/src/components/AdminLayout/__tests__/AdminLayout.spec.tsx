import * as React from 'react';
import { Navbar } from 'react-bootstrap';
import * as TestRenderer from 'react-test-renderer';
import { Sidebar } from '../../Sidebar';
import { AdminLayout } from '../AdminLayout';
import { AdminLayoutContent } from '../AdminLayoutContent';

test('renders a loading indicator with the default props', () => {
  const renderer = TestRenderer.create(<AdminLayout />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the layout when not loading', () => {
  const renderer = TestRenderer.create(<AdminLayout loading={false} />).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('does not render invalid children', () => {
  const renderer = TestRenderer.create(
    <AdminLayout loading={false}>
      <div>Awesome</div>
    </AdminLayout>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders a Sidebar when one is added as a child', () => {
  const renderer = TestRenderer.create(
    <AdminLayout loading={false}>
      <Sidebar dataColour="azure" backgroundColour="red" />
    </AdminLayout>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders a Navbar when one is added as a child', () => {
  const renderer = TestRenderer.create(
    <AdminLayout loading={false}>
      <Navbar />
    </AdminLayout>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders AdminLayoutContent when one is added as a child', () => {
  const renderer = TestRenderer.create(
    <AdminLayout loading={false}>
      <AdminLayoutContent />
    </AdminLayout>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});
