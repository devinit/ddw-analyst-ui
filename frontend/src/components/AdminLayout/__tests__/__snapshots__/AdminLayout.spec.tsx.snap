// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render invalid children 1`] = `
<div
  className="wrapper"
>
  <div
    className="main-panel"
  >
    <div
      className="content"
    >
      <div
        className="container-fluid"
      />
    </div>
  </div>
</div>
`;

exports[`renders AdminLayoutContent when one is added as a child 1`] = `
<div
  className="wrapper"
>
  <div
    className="main-panel"
  >
    <div
      className="content"
    >
      <div
        className="container-fluid"
      />
    </div>
  </div>
</div>
`;

exports[`renders a Navbar when one is added as a child 1`] = `
<div
  className="wrapper"
>
  <div
    className="main-panel"
  >
    <nav
      className="navbar navbar-expand navbar-light"
    />
    <div
      className="content"
    >
      <div
        className="container-fluid"
      />
    </div>
  </div>
</div>
`;

exports[`renders a Sidebar when one is added as a child 1`] = `
<div
  className="wrapper"
>
  <div
    className="sidebar"
    data-background-color="red"
    data-color="azure"
  />
  <div
    className="main-panel"
  >
    <div
      className="content"
    >
      <div
        className="container-fluid"
      />
    </div>
  </div>
</div>
`;

exports[`renders a loading indicator with the default props 1`] = `
<div>
  Loading ...
</div>
`;

exports[`renders the layout when not loading 1`] = `
<div
  className="wrapper"
>
  <div
    className="main-panel"
  >
    <div
      className="content"
    >
      <div
        className="container-fluid"
      />
    </div>
  </div>
</div>
`;
